fritz component test relies on missing mocker fixture

### The problem

The fritz config flow test has the following fixture, that relies upon the `mocker` fixture, which does not exist. It works when installing `pytest-mock`, but that package is not in the requirements.

https://github.com/home-assistant/core/blob/6df0190aeb944373c7abc15de04947b7fcc8fc68/tests/components/fritz/test_config_flow.py#L56-L58

Where is this fixture supposed to come from?

### What is version of Home Assistant Core has the issue?

2021.5.0

### What was the last working version of Home Assistant Core?

n/a

### What type of installation are you running?

Home Assistant Core

### Integration causing the issue

fritz

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/fritz

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

```txt
@pytest.fixture()
  def fc_class_mock(mocker):
E       fixture 'mocker' not found
>       available fixtures: aioclient_mock, aiohttp_client, aiohttp_raw_server, aiohttp_server, aiohttp_unused_port, bcrypt_cost, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, current_request, current_request_with_host, doctest_namespace, enable_custom_integrations, fail_on_log_exception, fast, fc_class_mock, hass, hass_access_token, hass_admin_credential, hass_admin_user, hass_client, hass_owner_user, hass_read_only_access_token, hass_read_only_user, hass_storage, hass_ws_client, legacy_auth, legacy_patchable_time, load_registries, local_auth, loop, loop_debug, mock_device_tracker_conf, mock_zeroconf, monkeypatch, mqtt_client_mock, mqtt_config, mqtt_mock, patch_zeroconf_multiple_catcher, prevent_io, proactor_loop, pytestconfig, raw_test_server, record_property, record_testsuite_property, record_xml_attribute, recwarn, requests_mock, stop_hass, test_client, test_server, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, unused_port, verify_cleanup, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.
```


### Additional information

_No response_

--- a/tests/components/fritz/test_config_flow.py
+++ b/tests/components/fritz/test_config_flow.py
@@ -54,11 +54,11 @@ MOCK_SSDP_DATA = {
 
 
 @pytest.fixture()
-def fc_class_mock(mocker):
+def fc_class_mock():
     """Fixture that sets up a mocked FritzConnection class."""
-    result = mocker.patch("fritzconnection.FritzConnection", autospec=True)
-    result.return_value = FritzConnectionMock()
-    yield result
+    with patch("fritzconnection.FritzConnection", autospec=True) as result:
+        result.return_value = FritzConnectionMock()
+        yield result
 
 
 async def test_user(hass: HomeAssistant, fc_class_mock):

